ui/src/main/webapp/components/FileInputComponent.tsx

```diff
@@ -40,10 +57,22 @@ function FileInputComponent(props) {
     const fileReader = new FileReader();
     const textDecoder = new TextDecoder(); // default utf-8
 
-    const [fileName, setFileName] = useState('');
-    const [errorMsg, setErrorMsg] = useState('');
+    /*
+     use fileNameToDisplay during editing to get
+     the possibility of removal previously added file
+    */
+    const [fileName, setFileName] = useState<string | null>(fileNameToDisplay || '');
 
-    const handleAddFiles = (files) => {
+    /* 
+      if the file data is encrypted and we display its name
+      then we display error message "file needs to be reuploaded"
+      as there is no access to data inside due to encription
+     */
+    const [errorMsg, setErrorMsg] = useState<JSX.Element | string>(
```

Comment:
JSX deprecated, use ReactElement from React instead

Reply:
i changed it into ReactElement